### ISSUE TITLE
Follow up items for #3019

### ISSUE THREAD

**Issue by crodriguezvega** [cosmos/ibc-go#3368]
Once #3019 is merged, these items will be worked on in a follow-up PR:

- [x] Write documentation (docs & migration) (https://github.com/cosmos/ibc-go/pull/3509)
- [x] Fix base transfer e2e test (https://github.com/cosmos/ibc-go/pull/3517).
- [x] Fix upgrade e2e test and make sure it passes (https://github.com/cosmos/ibc-go/pull/3605).
- [x] As recommended by @colin-axner, add [invariants](https://docs.cosmos.network/v0.47/building-modules/invariants) to check that actual amounts are equal or greater than the expected escrow amount. (https://github.com/cosmos/ibc-go/pull/3506)
- [x] Apply [this](https://github.com/cosmos/ibc-go/pull/3019#discussion_r1159974258) suggestion. (https://github.com/cosmos/ibc-go/pull/3507)
- [x] Add ADR and link in the code to the ADR. I think this should not block the release of the feature, though, and we can add it afterwards. (https://github.com/cosmos/ibc-go/pull/3641)
- [x] add link in migration docs once backport is merged (see [TODO](https://github.com/cosmos/ibc-go/pull/3509/files#diff-e158431e5ac177f77ad1bcc9310fa3bf2581214d053aa7c4b3ca80419d7b9fd3R48))

**Comment by crodriguezvega** [comment]
Closed by #3558
